pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -565,3 +602,61 @@ func (c *Reconciler) updateLabelsAndAnnotations(pr *v1alpha1.PipelineRun) (*v1al
 	}
 	return newPr, nil
 }
+
+func (c *Reconciler) makeConditionCheckContainer(logger *zap.SugaredLogger, rprt *resources.ResolvedPipelineRunTask, rcc *resources.ResolvedConditionCheck, pr *v1alpha1.PipelineRun) (*v1alpha1.ConditionCheck, error) {
+	labels := getTaskrunLabels(pr, rprt.PipelineTask.Name)
+	labels[pipeline.GroupName+pipeline.PipelineRunConditionCheckKey] = rcc.ConditionCheckName
+
+	tr := &v1alpha1.TaskRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            rcc.ConditionCheckName,
+			Namespace:       pr.Namespace,
+			OwnerReferences: pr.GetOwnerReference(),
+			Labels:          labels,
+			Annotations:     getTaskrunAnnotations(pr), // Propagate annotations from PipelineRun to TaskRun.
+		},
+		Spec: v1alpha1.TaskRunSpec{
+			TaskSpec:       rcc.ConditionToTaskSpec(),
+			ServiceAccount: getServiceAccount(pr, rprt.PipelineTask.Name),
+			Timeout:        getTaskRunTimeout(pr),
+			NodeSelector:   pr.Spec.NodeSelector,
+			Tolerations:    pr.Spec.Tolerations,
+			Affinity:       pr.Spec.Affinity,
+		}}
+
+	cctr, err := c.PipelineClientSet.TektonV1alpha1().TaskRuns(pr.Namespace).Create(tr)
+	cc := v1alpha1.ConditionCheck(*cctr)
+	return &cc, err
+}
+
+func getTaskRunTimeout(pr *v1alpha1.PipelineRun) *metav1.Duration {
+	var taskRunTimeout = &metav1.Duration{Duration: 0 * time.Second}
+
+	if pr.Spec.Timeout != nil {
```

Comment:
with https://github.com/tektoncd/pipeline/pull/1053 (and also https://github.com/tektoncd/pipeline/pull/1040), `Timeout` should never be nil.